xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-livetable/src/main/java/org/xwiki/livedata/internal/livetable/DefaultLiveDataConfigurationResolver.java

```diff
@@ -199,12 +199,26 @@ private LiveDataPropertyDescriptor getDefaultPropertyDescriptor(String property)
         return propertyDescriptor;
     }
 
-    private LiveDataConfiguration translate(LiveDataConfiguration config)
+    /**
+     * Updates the {@code mergedConfig} with translated property names and descriptions.
+     *
+     * @param mergedConfig the configuration to update
+     * @param config the base configuration of the live data macro
+     * @return the updated {@code mergedConfig}
+     */
+    private LiveDataConfiguration translate(LiveDataConfiguration mergedConfig, LiveDataConfiguration config)
     {
-        String translationPrefix = (String) config.getQuery().getSource().getParameters().get("translationPrefix");
-        for (LiveDataPropertyDescriptor property : config.getMeta().getPropertyDescriptors()) {
-            if (property.getName() == null) {
-                property.setName(this.l10n.getTranslationPlain(translationPrefix + property.getId()));
+        String translationPrefix =
+            (String) mergedConfig.getQuery().getSource().getParameters().get("translationPrefix");
+        for (LiveDataPropertyDescriptor property : mergedConfig.getMeta().getPropertyDescriptors()) {
+            // We set a name only if the property does not already have a name or if it does not have a default 
+            // name.
+            if (property.getName() == null || !hasDefaultName(config, property.getId())) {
```

Comment:
```suggestion
            // If the property name is not set then we default on the configured translation key or the property id.
            // Otherwise, if the property name is set but not by the user (i.e. the name comes from the default source
            // configuration) then we want to give priority to the configured translation key if available.
            if (property.getName() == null || !hasSpecifiedName(config, property.getId())) {
```